app/socket.ts

```diff
@@ -1,6 +1,7 @@
 "use client";
 
-import { io, Socket } from "socket.io-client";
 import type { ClientToServerEvents, ServerToClientEvents } from "@/types";
+import { io, Socket } from "socket.io-client";
 
-export const socket: Socket<ServerToClientEvents, ClientToServerEvents> = io();
+export type SocketType = Socket<ServerToClientEvents, ClientToServerEvents>;
```

Comment:
no need to export this type, just import `socket` where you need it, even in hooks